### PR TITLE
Feat/profiling fixes

### PR DESCRIPTION
**Pull request by stefanosiano**
## :scroll: Description

reduced profiling sampling rate from 300hz to 100hz
profiling trace folder is now created regardless of profiling enabled option

## :bulb: Motivation and Context
Manual initialisation happens after profiling traces folder creation, so we were losing profiling when it was set manually, as the traces folder was not being created

## :green_heart: How did you test it?
Running a sample with strict mode did not complain about folder creation

## :pencil: Checklist

- [ ] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] No breaking changes

### CODE REVIEW

File: sentry-android-core/src/main/java/io/sentry/android/core/AndroidOptionsInitializer.java [hunk 1]
```diff
@@ -305,23 +305,23 @@ private static void initializeCacheDirs(
...
-                if (oldTracesDirContent == null) return;
-                // Method trace files are normally deleted at the end of traces, but if that fails
-                // for some
-                // reason we try to clear any old files here.
-                for (File f : oldTracesDirContent) {
-                  FileUtils.deleteRecursively(f);
-                }
-              });
-    }
+    // We don't have the options set by code (manual initialisation) at this point, so we have no
```
Comment (marandaneto) [inline sentry-android-core/src/main/java/io/sentry/android/core/AndroidOptionsInitializer.java:308]: That's why we do the file handling in the Sentry init class
https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/Sentry.java#L222-L229=
Can we move this logic in there? so we don't need create a folder unnecessarily

Comment (stefanosiano) [reply sentry-android-core/src/main/java/io/sentry/android/core/AndroidOptionsInitializer.java:308]: alright, but i should move `profilingTracesDirPath` from `SentryAndroidOptions` to `SentryOptions` (`profilingEnabled` is already in `SentryOptions`).
Is it fine?

File: sentry-android-core/src/test/java/io/sentry/android/core/AndroidTransactionProfilerTest.kt [hunk 2]
```diff
@@ -46,6 +49,7 @@ class AndroidTransactionProfilerTest {
         context = ApplicationProvider.getApplicationContext()
         file = context.cacheDir
         AndroidOptionsInitializer.init(fixture.options, createMockContext())
+        Sentry.init(fixture.options)
```
Comment (marandaneto) [inline sentry-android-core/src/test/java/io/sentry/android/core/AndroidTransactionProfilerTest.kt:52]: Not all the tests require `Sentry.init`, so I'd rather do in a specific test and also call `Sentry.close` after.

Comment (stefanosiano) [reply sentry-android-core/src/test/java/io/sentry/android/core/AndroidTransactionProfilerTest.kt:52]: yeah, it's needed to generate the profiling trace folder. Was missing the `Sentry.close` part 😅

**Comment by marandaneto** [comment]
@romtsn would you like to do a final review? :)

File: sentry/src/main/java/io/sentry/Sentry.java [hunk 3]
```diff
@@ -231,6 +233,28 @@ private static boolean initConfigurations(final @NotNull SentryOptions options)
       options.setEnvelopeDiskCache(EnvelopeCache.create(options));
     }
 
+    String profilingTracesDirPath = options.getProfilingTracesDirPath();
```
Comment (romtsn) [inline sentry/src/main/java/io/sentry/Sentry.java:236]:
```suggestion
    final String profilingTracesDirPath = options.getProfilingTracesDirPath();
```

**Review by romtsn — APPROVED** [review]

**Pull request merged**